CMake Error on Cygwin (3.3.6, current version): Policy "CMP0135" is not known

The newest version of cmake in Cygwin is 3.23.2-1, but [CMP0135](https://cmake.org/cmake/help/latest/policy/CMP0135.html#policy:CMP0135) was not added until 3.24.

```
$ cmake -DTARGET_ARCH=x86_64-w64-mingw32 -G Ninja ..
CMake Error at CMakeLists.txt:9 (cmake_policy):
  Policy "CMP0135" is not known to this version of CMake.
```

In #243 the new policy was added. Can you maybe add a condition, so that it is possible to build in Cygwin?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,9 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/custom_steps.cmake)
 
 cmake_policy(SET CMP0097 NEW)
 cmake_policy(SET CMP0114 NEW)
-cmake_policy(SET CMP0135 NEW)
+if(POLICY CMP0135)
+  cmake_policy(SET CMP0135 NEW)
+endif()
 
 if(NOT CMAKE_GENERATOR STREQUAL "Ninja")
     message(WARNING "Generator “${CMAKE_GENERATOR}” is unsupported!\nTry Ninja if you encounter problems.")

